### ISSUE TITLE
Feature Request ->  WORDPRESS_DATABASE_PASSWORD_FILE

### ISSUE THREAD

**Issue by stunney** [bitnami/bitnami-docker-wordpress-nginx#24]
The Bitnami MariaDb container allows one to pass in the ROOT and default-created user's passwords in as secrets in a file with Docker-Compose.

Example for MariaDb:
```
version: '3.6'
services:
  mariadb:
    image: 'bitnami/mariadb:10.4'
    ports:
      - '3306:3306'
    secrets:
      - mysql_r
      - mysql_wp
    environment:
      - MARIADB_DATABASE=bitnami_wordpress
      - MARIADB_USER=bn_wordpress
      - MARIADB_PASSWORD_FILE=/run/secrets/mysql_wp
      - MARIADB_ROOT_PASSWORD_FILE=/run/secrets/mysql_r
secrets:
  mysql_r:
    file: ../bitnami/environments/${DEPLOY_STAGE}/secrets/MYSQL_ROOT_PASSWORD
  mysql_wp:
    file: ../bitnami/environments/${DEPLOY_STAGE}/secrets/MYSQL_WORDPRESS_PASSWORD

```

Please add a similar environment variable for the wordpress-nginx container so that it can also take advantage of secrets, properly.

WORDPRESS_DATABASE_PASSWORD_FILE

Example:
```
  wordpress:
    image: 'bitnami/wordpress-nginx:5.4.1'
    ports:
      - '80:8080'
      - '443:8443'
    secrets:
      - mysql_wp
    depends_on:
      - mariadb
    links:
      - mariadb
    environment:
      - MARIADB_HOST=mariadb
      - MARIADB_PORT_NUMBER=3306
      - WORDPRESS_DATABASE_USER=bn_wordpress
      - WORDPRESS_DATABASE_PASSWORD_FILE=/run/secrets/mysql_wp
      - WORDPRESS_DATABASE_NAME=bitnami_wordpress

```

**Comment by stunney** [comment]
And I tried using a .env file, and reading it in as a following example:
```
version: '3.6'
services:
  mariadb:
    image: 'bitnami/mariadb:10.4'
    env_file:
      - ./.env
    ports:
      - '3306:3306'
    environment:
      - MARIADB_DATABASE=bitnami_wordpress
      - MARIADB_USER=bn_wordpress
      - MARIADB_PASSWORD=${WP_DB_PW}
      - MARIADB_ROOT_PASSWORD=${WP_R_PW}
    networks:
      - app-network
  wordpress:
    image: 'bitnami/wordpress-nginx:5.4.1'
    env_file:
      - ./.env
    ports:
      - '80:8080'
      - '443:8443'
    depends_on:
      - mariadb
    links:
      - mariadb
    command:
      - 'echo ${WP_DB_PW}'
    environment:
      - MARIADB_HOST=mariadb
      - MARIADB_PORT_NUMBER=3306
      - WORDPRESS_DATABASE_USER=bn_wordpress
      - **WORDPRESS_DATABASE_PASSWORD**=${WP_DB_PW}
      - WORDPRESS_DATABASE_NAME=bitnami_wordpress
networks:
  app-network:
    driver: bridge
```

.env (scrubbed)
```
DOMAIN=mydomain
WP_DB_PW=superreallylongpasswordthatisnotlikelytobeguessedbyanyone
WP_R_PW=superreallylongpasswordthatisnotlikelytobeguessedbyanyonerootrules

```

And the wordpress-nginx container comes back with the following weird error:
```
wordpress_1  |
wordpress_1  | Welcome to the Bitnami wordpress-nginx container
wordpress_1  | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-wordpress-nginx
wordpress_1  | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-wordpress-nginx/issues
wordpress_1  |
wordpress_1  | [FATAL tini (8)] exec echo superreallylongpasswordthatisnotlikelytobeguessedbyanyone failed: No such file or directory
mariadb_wordpress_1 exited with code 127
```

Why is it trying to treat the password as a file?  That is so odd.

**Comment by dani8art** [comment]
Hi @stunney , thank you for the suggestion, it could be interesting to support this feature, it is cool. We have opened an internal task in order to evaluate it, unfortunately, due to there is a way to pass custom passwords to the container and regarding priorities and internal milestones we can't give you an ETA. However, if it is important for you and you feel you can handle it and you have a chance to create a PR we would be happy to handle it. 

I can give you some tips for managing this:

Wordpress configuration is handle by a `nami` module. You can find the configuration scripts and the `nami`'s files here:

```console
ls -la .nami/components/com.bitnami.wordpress/
```

The main components of a `nami` module are:
- `.nami/components/com.bitnami.wordpress/main.js` where the logic is implemented, this is executed at container runtime and will configure the Wordpress instance.
- `.nami/components/com.bitnami.wordpress/nami.json` where some metadata and properties are stored. 
- `/wordpress-inputs.json` this file maps the environment variables to `nami` module properties.

So if you want to add a new property you would need to do the following:

1. Add a new property to `nami.json`, e.g. `databasePasswordFile`
2. Add logic in `main.js` as if `$app.databasePasswordFile` has value then load the password from this file path. There are some utils modules in `nami` like `$file`, `$os`, etc... 
3. Add a mapping between env vars and `nami` properties in `wordpress-inputs.json`

I hope you are finally encouraged to do it.

**Comment by marcosbc** [comment]
Hi @stunney, this was just implemented in [`5.7.1-debian-10-r22`](https://github.com/bitnami/bitnami-docker-wordpress-nginx#571-debian-10-r22), so I'm proceeding to closing this issue.
